esmvaltool/preprocessor/_mask.py

```diff
@@ -17,20 +18,107 @@
 logger = logging.getLogger(__name__)
 
 
-def mask_landocean(cube, land_mask=None, ocean_mask=None):
-    """Mask land or ocean"""
-    if land_mask or ocean_mask:
-        raise NotImplementedError
-
+def _check_dims(cube, mask_cube):
+    """Check for same dims for mask and data"""
+    x_dim = cube.coord('longitude').points.ndim
+    y_dim = cube.coord('latitude').points.ndim
+    mx_dim = mask_cube.coord('longitude').points.ndim
+    my_dim = mask_cube.coord('latitude').points.ndim
+    len_x = len(cube.coord('longitude').points)
+    len_y = len(cube.coord('latitude').points)
+    len_mx = len(mask_cube.coord('longitude').points)
+    len_my = len(mask_cube.coord('latitude').points)
+    if x_dim == mx_dim and y_dim == my_dim:
+        if len_x == len_mx and len_y == len_my:
+            logger.debug('Data cube and fx mask have same dims')
+            return True
+        else:
+            logger.error('Data cube and fx mask have different grids!')
```

Comment:
This is probably a problem with the `logger.error`, but when the code gets here, it prints the error string, but then later in the tracerback it gives:
```
UnboundLocalError: local variable 'landsea_mask' referenced before assignment
```
@bouweandela why is it so?

Reply:
because the error doesn't actually stop the code, it goes on with ```landsea_mask``` undefined (hence the UnboundLocalError) and crashes there

Reply:
@mattiarighi could you send over the stack trace pls?

Reply:
crap, hang on, I know what the problem is --

Reply:
ok @mattiarighi fixed it, you should be OK now, man, how many corner cases are there? :grinning: